Source/Particles/PhysicalParticleContainer.H

```diff
@@ -280,6 +280,15 @@ public:
         amrex::FArrayBox const * & ezfab, amrex::FArrayBox const * & bxfab,
         amrex::FArrayBox const * & byfab, amrex::FArrayBox const * & bzfab);
 
+    /**
+    * \brief This function determines if resampling should be done for the current species, and
+    * if so, performs the resampling.
+    *
+    * @param[in] resampler the Resampling object used for resampling.
+    * @param[in] timestep the current timestep.
+    */
+    void resample (const Resampling& resampler, const int timestep) override;
```

Comment:
Can this also be final?
```suggestion
    void resample (const Resampling& resampler, const int timestep) override final;
```

Reply:
I think it can yes. I've also put the `LevelingThinning` `operator()` as final.